### ISSUE TITLE
Update for gcc version 11 needed

### ISSUE THREAD

**Issue by makingglitches** [bazelbuild/bazel#13635]
https://github.com/bazelbuild/bazel/issues/13613

 Code downloaded during build process com_google package requires either manual editing moving header references to header files if the symbols are being referenced or gcc 10 may need to be used

Adding a rule with preprocessing using symbol defines as part of compiler version detection during configuration could solve this but code edits would have to happen in any case 

Also limits.h from stl needs to be changed to limits without extension in includes for stl 

Suggested method is simply defining a symbol -DGCC11 which preprocessing could use to move the header file inclusions to the proper file 

Where are the externs coming from is part of the information that needs provided as this might require some fixes across a few projects if they are not maintained in this one.

**Comment by makingglitches** [comment]
Props to @neilg

May have discovered a broader issue than that in

https://github.com/bazelbuild/bazel/issues/13613

**Comment by makingglitches** [comment]
workaround exists according to:

https://github.com/bazelbuild/bazel/issues/13613#issuecomment-872400814
for some distros.
probably just a matter of running two parallel versions of gcc on fedora, which I don't feel personally is all that great an idea.

seems like it might carry with it some issues, would be better in my humble opinion just to make the edits as proscribed to detect the gcc version at configure-time and modify failing code files to respond to the idea mentioned above.

**Comment by davido** [comment]
This is a duplicate of https://github.com/bazelbuild/bazel/issues/12756.

Also note, that this problem is fixed with this PR: https://github.com/bazelbuild/bazel/pull/13536,
but it's blocked on https://github.com/bazelbuild/bazel/issues/13561.

**Comment by makingglitches** [comment]
@davido what is the source of the externs if you don't mind me asking ? Are they being maintained as some connected project ?
